hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientReplicatedMapProxy.java

```diff
@@ -112,7 +115,7 @@ protected void onInitialize() {
     private void initNearCache() {
         NearCacheConfig nearCacheConfig = getContext().getClientConfig().getNearCacheConfig(name);
         if (nearCacheConfig != null) {
-            nearCache = getContext().getNearCacheManager().getOrCreateNearCache(name, nearCacheConfig);
+            nearCache = getContext().getNearCacheManager().getOrCreateNearCache(nearCacheIdentifier, nearCacheConfig);
```

Comment:
I think we don't need a new class like `NearCacheIdentifier` since we already have unique names 
that are coming from proxy names, we only need `NearCacheManager` per service which can be implemented as in [here]( https://github.com/hazelcast/hazelcast/blob/5ddc7e7e4715a0bae0efe49a9d5b6a2945689d73/hazelcast/src/main/java/com/hazelcast/client/impl/spi/ClientContext.java#L95). So when you give service name you can get matching `NearCacheManager` for that service. 
Would you mind to rewrite your fix based on the referred link?

Reply:
updated, squashed and rebased over master branch